Nodejs/Product/Nodejs/TestFrameworks/TestFrameworkDirectories.cs

```diff
@@ -1,53 +1,48 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
 
 namespace Microsoft.NodejsTools.TestFrameworks
 {
-    internal class TestFrameworkDirectories
+    internal static class TestFrameworkDirectories
     {
         public const string ExportRunnerFrameworkName = "ExportRunner";
         private const string TestFrameworksFolderName = "TestFrameworks";
         private const string TestAdapterFolderName = "TestAdapter";
 
-        private readonly Dictionary<string, string> frameworkDirectories = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-
-        public static string TestFrameworksFolder1 => TestFrameworksFolderName;
-
-        public TestFrameworkDirectories()
+        public static string[] GetFrameworkNames()
         {
             var testFrameworkRoot = GetTestframeworkFolderRoot();
             if (!Directory.Exists(testFrameworkRoot))
             {
                 throw new InvalidOperationException($"Unable to find test framework folder. Tried: \"{testFrameworkRoot}\"");
             }
 
-            foreach (var directory in Directory.EnumerateDirectories(testFrameworkRoot))
-            {
-                var name = Path.GetFileName(directory);
-                this.frameworkDirectories.Add(name, directory);
-            }
+            return Directory.EnumerateDirectories(testFrameworkRoot).Select(dir => Path.GetFileName(dir)).ToArray();
```

Comment:
>return Directory.EnumerateDirectories(testFrameworkRoot).Select(dir => Path.GetFileName(dir)).ToArray(); [](start = 12, length = 104)

Alternatively, `return GetFrameworkDirectories().Select(Path.GetFileName).ToArray();`.

Reply:
Getting rid of the lambda is nice, but my point was actually that the method would be a one-liner if it delegated to `GetFrameworkDirectories`.

---
In reply to: [186890964](https://github.com/Microsoft/nodejstools/pull/1937#discussion_r186890964) [](ancestors = 186890964)